aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -0,0 +1,2326 @@
+        // but adding rewards onto the existing stake is still a good approximation
+        if (delegator_address == stake::get_operator(pool_address)) {
+            active = active + commission_active;
+            // in-flight pending_inactive commission can coexist with already inactive withdrawal
+            if (lockup_cycle_ended) {
+                inactive = inactive + commission_pending_inactive
+            } else {
+                pending_inactive = pending_inactive + commission_pending_inactive
+            }
+        };
+
+        (active, inactive, pending_inactive)
+    }
+
+    #[view]
+    /// Return refundable stake to be extracted from added `amount` at `add_stake` operation on pool `pool_address`.
+    /// If the validator produces rewards this epoch, added stake goes directly to `pending_active` and
+    /// does not earn rewards. However, all shares within a pool appreciate uniformly and when this epoch ends:
+    /// - either added shares are still `pending_active` and steal from rewards of existing `active` stake
+    /// - or have moved to `pending_inactive` and get full rewards (they displaced `active` stake at `unlock`)
+    /// To mitigate this, some of the added stake is extracted and fed back into the pool as placeholder
+    /// for the rewards the remaining stake would have earned if active:
+    /// extracted-fee = (amount - extracted-fee) * reward-rate% * (100% - operator-commission%)
+    public fun get_add_stake_fee(pool_address: address, amount: u64): u64 acquires DelegationPool {
+        if (stake::is_current_epoch_validator(pool_address)) {
+            let (rewards_rate, rewards_rate_denominator) = staking_config::get_reward_rate(&staking_config::get());
+            if (rewards_rate_denominator > 0) {
+                assert_delegation_pool_exists(pool_address);
+                let pool = borrow_global<DelegationPool>(pool_address);
+
+                rewards_rate = rewards_rate * (MAX_FEE - pool.operator_commission_percentage);
+                rewards_rate_denominator = rewards_rate_denominator * MAX_FEE;
+                ((((amount as u128) * (rewards_rate as u128)) / ((rewards_rate as u128) + (rewards_rate_denominator as u128))) as u64)
+            } else { 0 }
+        } else { 0 }
+    }
+
+    #[view]
+    /// Return whether `pending_inactive` stake can be directly withdrawn from
+    /// the delegation pool, implicitly its stake pool, in the special case
+    /// the validator had gone inactive before its lockup expired.
+    public fun can_withdraw_pending_inactive(pool_address: address): bool {
+        stake::get_validator_state(pool_address) == VALIDATOR_STATUS_INACTIVE &&
+        timestamp::now_seconds() >= stake::get_lockup_secs(pool_address)
+    }
+
+    public entry fun initialize_delegation_pool(
+        owner: &signer,
+        operator_commission_percentage: u64,
+        delegation_pool_creation_seed: vector<u8>,
+    ) {
```

Comment:
Can you gate this function with a feature flag, similar to https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/stake.move#L1177. This would allow us to separate actually enabling delegation from the framework upgrade